### PR TITLE
fix(components): [cascader] modelValue update problem

### PR DESCRIPTION
**Pull request by chenxch**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by chenxch** [comment]
[before](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZXhhbXBsZS1ibG9ja1wiPlxuICAgIDxzcGFuIGNsYXNzPVwiZXhhbXBsZS1kZW1vbnN0cmF0aW9uXCI+RGlzcGxheSBhbGwgdGFncyAoZGVmYXVsdCk8L3NwYW4+XG4gICAgPGJ1dHRvbiBAY2xpY2s9XCJoYW5kbGVEZWxldGVcIj7liKDpmaTkuIDkuKo8L2J1dHRvbj5cbiAgICB7e2xpc3R9fVxuICAgIDxicj5cbiAgICA8ZWwtY2FzY2FkZXIgdi1tb2RlbD1cImxpc3RcIiA6b3B0aW9ucz1cIm9wdGlvbnNcIiA6cHJvcHM9XCJwcm9wc1wiIGNsZWFyYWJsZSAvPlxuICAgIDxlbC1jYXNjYWRlci1wYW5lbCB2LW1vZGVsPVwibGlzdFwiIDpvcHRpb25zPVwib3B0aW9uc1wiIDpwcm9wcz1cInByb3BzXCIgLz5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSBcInZ1ZVwiXG5jb25zdCBwcm9wcyA9IHsgbXVsdGlwbGU6IHRydWUsIGVtaXRQYXRoOiBmYWxzZSB9XG5jb25zdCBsaXN0ID0gcmVmKFszLDQsNV0pXG5mdW5jdGlvbiBoYW5kbGVEZWxldGUoKSB7XG4gIGxpc3QudmFsdWUucG9wKCk7IC8vIOayoeacieeUqFxuICAvLyBsaXN0ID0gW10g5pyJ55SoXG4gIGNvbnNvbGUubG9nKGxpc3QudmFsdWUpO1xufVxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIHZhbHVlOiAxLFxuICAgIGxhYmVsOiAnQXNpYScsXG4gICAgY2hpbGRyZW46IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6IDIsXG4gICAgICAgIGxhYmVsOiAnQ2hpbmEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDMsIGxhYmVsOiAnQmVpamluZycgfSxcbiAgICAgICAgICB7IHZhbHVlOiA0LCBsYWJlbDogJ1NoYW5naGFpJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDUsIGxhYmVsOiAnSGFuZ3pob3UnIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogNixcbiAgICAgICAgbGFiZWw6ICdKYXBhbicsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogNywgbGFiZWw6ICdUb2t5bycgfSxcbiAgICAgICAgICB7IHZhbHVlOiA4LCBsYWJlbDogJ09zYWthJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDksIGxhYmVsOiAnS3lvdG8nIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogMTAsXG4gICAgICAgIGxhYmVsOiAnS29yZWEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDExLCBsYWJlbDogJ1Nlb3VsJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDEyLCBsYWJlbDogJ0J1c2FuJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDEzLCBsYWJlbDogJ1RhZWd1JyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgdmFsdWU6IDE0LFxuICAgIGxhYmVsOiAnRXVyb3BlJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICB2YWx1ZTogMTUsXG4gICAgICAgIGxhYmVsOiAnRnJhbmNlJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7IHZhbHVlOiAxNiwgbGFiZWw6ICdQYXJpcycgfSxcbiAgICAgICAgICB7IHZhbHVlOiAxNywgbGFiZWw6ICdNYXJzZWlsbGUnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMTgsIGxhYmVsOiAnTHlvbicgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAxOSxcbiAgICAgICAgbGFiZWw6ICdVSycsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogMjAsIGxhYmVsOiAnTG9uZG9uJyB9LFxuICAgICAgICAgIHsgdmFsdWU6IDIxLCBsYWJlbDogJ0Jpcm1pbmdoYW0nIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMjIsIGxhYmVsOiAnTWFuY2hlc3RlcicgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAyMyxcbiAgICBsYWJlbDogJ05vcnRoIEFtZXJpY2EnLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAyNCxcbiAgICAgICAgbGFiZWw6ICdVUycsXG4gICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgeyB2YWx1ZTogMjUsIGxhYmVsOiAnTmV3IFlvcmsnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMjYsIGxhYmVsOiAnTG9zIEFuZ2VsZXMnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMjcsIGxhYmVsOiAnV2FzaGluZ3RvbicgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAyOCxcbiAgICAgICAgbGFiZWw6ICdDYW5hZGEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHsgdmFsdWU6IDI5LCBsYWJlbDogJ1Rvcm9udG8nIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMzAsIGxhYmVsOiAnTW9udHJlYWwnIH0sXG4gICAgICAgICAgeyB2YWx1ZTogMzEsIGxhYmVsOiAnT3R0YXdhJyB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG5cbjxzdHlsZSBzY29wZWQ+XG4uZXhhbXBsZS1ibG9jayB7XG4gIG1hcmdpbjogMXJlbTtcbn1cbi5leGFtcGxlLWRlbW9uc3RyYXRpb24ge1xuICBtYXJnaW46IDFyZW07XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

**Comment by chenxch** [comment]
> https://github.com/element-plus/element-plus/blob/f6157c73125fc80a1398a3cad0532f8fb730fcf9/packages/components/cascader-panel/src/index.vue#L213-L218
> 
> 
> This `isEqual` seems to have no effect, because it will always be `false`.

ignore

**Review by jw-foss — COMMENTED** [review]
@chenxch I left a comment for your considerations, otherwise this PR LGTM, let me know if you have any questions, back to you :ping_pong:

File: packages/components/cascader-panel/src/index.vue [hunk 1]
```diff
@@ -238,7 +239,9 @@ export default defineComponent({
           values.map((val) => store?.getNodeByValue(val, leafOnly))
         ) as Node[]
         syncMenuState(nodes, forced)
-        checkedValue.value = modelValue!
+        checkedValue.value = isArray(modelValue)
+          ? [...modelValue]
+          : { ...modelValue }
```
[the remark this answers is not in the document]
Comment (chenxch) [reply packages/components/cascader-panel/src/index.vue:244]: Good idea, so it's used to differentiate the types.

**Review by jw-foss — APPROVED** [review]
@chenxch thanks, the changes LGTM now. Approving.

**Pull request merged**